### ISSUE TITLE
isconstant method in Di/1 runs into an error when creating beanfactories manually.

### ISSUE THREAD

**Issue by Ronner** [framework-one/fw1#315]
When I was doing some testing to make the beans of the default subsystem available to other subsystems, I ran into something that I thought I'd create an issue for.

I applied the following code;

```cfc
variables.framework = {
	base = '/mybase/',
	usingSubsystems = true,
	defaultSubsystem = 'core',
	reloadApplicationOnEveryRequest = true,
	generateSES = true,
	SESOmitIndex = true,
	diEngine='none'
};

function setupApplication() {
		
	// Create beanfactory manually to make the core subsystems beans available to other subsystems (see setupSubsystem method)
	local.folders = [
		variables.framework.base & variables.framework.defaultSubsystem & '/controllers'
		, variables.framework.base & variables.framework.defaultSubsystem & '/model'
	];
	
	local.bf = new framework.ioc(arrayToList(local.folders), { 
		initMethod = 'postconstruct',
		omitDirectoryAliases = true
	});

	local.bf.addBean("fw", this);
	setBeanFactory(local.bf);
}

function setupSubsystem(subsystem) {
	
	// For each subsystem, create a new beanfactory and make the defaultBeanFactory it's parent, so that beans from the core
	// application are available to subsystems
	local.folders = [
		variables.framework.base & arguments.subsystem & '/controllers'
		, variables.framework.base & arguments.subsystem & '/model'
	];
	
	local.bf = new framework.ioc(arrayToList(local.folders), { 
		initMethod = 'postconstruct',
		omitDirectoryAliases = true
	});

	local.bf.setParent(getDefaultBeanFactory());
	setSubsystemBeanFactory(arguments.subsystem, local.bf);
}
```

When using this, di/1 throws an error around line 475 in the isconstant method. When I change DI/1's code inside the isconstant method from;

```cfc
return structKeyExists( variables.beanInfo[ beanName ], 'value');
```

to:

```cfc
return structKeyExists(variables.beanInfo, beanName) && structKeyExists( variables.beanInfo[ beanName ], 'value');
```

It seems to work fine, but I have not had the time to investigate if this is a proper fix for this problem.

**Comment by seancorfield** [comment]
The call to `isConstant()` should probably verify `containsBean()` first but `isConstant()` should probably respect parent bean factories too. Marking for 3.1.

**Comment by Ronner** [comment]
Is there a roadmap yet for 3.1?

**Comment by seancorfield** [comment]
https://github.com/framework-one/fw1/milestones/3.1 -- subject to change... :)

**Comment by seancorfield** [comment]
Fixed in 3.1 develop.
